resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/DepsTestsSuite.scala

```diff
@@ -0,0 +1,237 @@
+    val envVars = Map ( "NETWORK_AUTO_DETECT" -> "4",
+      "RGW_CIVETWEB_PORT" -> "8000",
+      "SREE_PORT" -> "5001",
+      "CEPH_DEMO_UID" -> "nano",
+      "CEPH_DAEMON" -> "demo",
+      "DEBUG" -> "verbose"
+    ).map( envV =>
+      new EnvVarBuilder()
+        .withName(envV._1)
+        .withValue(envV._2)
+        .build()
+    ).toArray
+
+    val resources = Map(
+      "cpu" -> new QuantityBuilder()
+        .withAmount("1")
+        .build(),
+      "memory" -> new QuantityBuilder()
+        .withAmount("512M")
+        .build()
+    ).asJava
+
+    new ContainerBuilder()
+      .withImage("ceph/daemon")
+      .withImagePullPolicy("Always")
+      .withName(cName)
+      .withPorts(new ContainerPortBuilder()
+          .withName(svcName)
+          .withProtocol("TCP")
+          .withContainerPort(8000)
+        .build()
+      )
+      .withResources(new ResourceRequirementsBuilder()
+        .withLimits(resources)
+        .withRequests(resources)
+        .build()
+      )
+      .withEnv(envVars: _*)
+      .build()
+  }
+
+  // Based on https://github.com/ceph/cn
+  private def setupCephStorage(): Unit = {
+    val labels = Map("app" -> "ceph", "daemon" -> "nano").asJava
+    val cephService = new ServiceBuilder()
+      .withNewMetadata()
+        .withName(svcName)
+      .withLabels(labels)
+      .endMetadata()
+      .withNewSpec()
+        .withPorts(new ServicePortBuilder()
+          .withName("https")
+          .withPort(8000)
+          .withProtocol("TCP")
+          .withTargetPort(new IntOrString(8000))
+          .build()
+        )
+        .withType("NodePort")
+        .withSelector(labels)
+      .endSpec()
+      .build()
+
+    val cephStatefulSet = new StatefulSetBuilder()
+      .withNewMetadata()
+        .withName(cName)
+        .withLabels(labels)
+      .endMetadata()
+      .withNewSpec()
+        .withReplicas(1)
+        .withNewSelector()
+          .withMatchLabels(Map("app" -> "ceph").asJava)
+        .endSelector()
+        .withServiceName(cName)
+        .withNewTemplate()
+          .withNewMetadata()
+            .withName(cName)
+            .withLabels(labels)
+           .endMetadata()
+          .withNewSpec()
+            .withContainers(getCephContainer())
+          .endSpec()
+        .endTemplate()
+      .endSpec()
+      .build()
+
+    kubernetesTestComponents
+      .kubernetesClient
+      .services()
+      .create(cephService)
+
+    kubernetesTestComponents
+      .kubernetesClient
+      .apps()
+      .statefulSets()
+      .create(cephStatefulSet)
+  }
+
+ private def deleteCephStorage(): Unit = {
+    kubernetesTestComponents
+      .kubernetesClient
+      .apps()
+      .statefulSets()
+      .withName(cName)
+      .delete()
+
+    kubernetesTestComponents
+      .kubernetesClient
+      .services()
+      .withName(svcName)
+      .delete()
+  }
+
+  test("Test launcher client dependencies", k8sTestTag, MinikubeTag) {
```

Comment:
Pet peeve of mine, but including "Test" in the test description is kinda redundant.

Reply:
ok.